### ISSUE TITLE
Grouping Legend by Axis

### ISSUE THREAD

**Issue by axg1010** [oxyplot/oxyplot#643]
When using multiple axes, it can be hard to discern which series belong to which axis. I've added a property (LegendGrouping) to PlotModel which can be set to group by X-Axis, Y-Axis, or None (works as before). If set to group (either by X or Y axis), each entry in the legend appears under the name of it's respective axis (one heading per axis). I've used and appreciated OxyPlot for a while now. Personally this is a feature I'm in need of and hope that others will find it useful too.

Disclaimer:
I am new to Git and contributing to open source projects in general. I'm attempting to follow the procedure [here](https://github.com/oxyplot/oxyplot/blob/develop/CONTRIBUTING.md). Although I've already implemented this feature on my local machine, I believe I am suppose to create this issue and wait for it to be tagged in order for me to create a pull request. Please let me know if this is not correct.

**Comment by objorke** [comment]
Good idea! But I think we need some discussion here! I remember there has been some other discussion about legends - like adding multiple legend boxes (then you could show all legends related to an axis next to the axis). This requires some refactoring, at least in the layout (measure/arrange) engine. I think the first step could be to add a `Legends` property (a collection like `Axes` and `Series`) to the `PlotModel`. All the current legend properties should then be moved into a default implementation of a "legend box" (breaking change)... This should make the legends extendable and it should be implement grouping as you suggest!

Yes, please follow the contributing guidelines! Would you like to take the challenge to refactor the legend architecture?

**Comment by axg1010** [comment]
My original implementation attempted to avoid "breaking changes" but if that is not a concern, your proposed plan would add more flexible functionality. So yes, I would be willing to take on this task.

**Comment by objorke** [comment]
For such refactoring I think breaking changes are ok. I will add more details on #644, and assign the issue to you! This issue should be solved after #644.
